src/Xrefcheck/Verify.hs

```diff
@@ -269,14 +273,24 @@ verifyReference
                 givenAnchors
           in throwError $ AnchorDoesNotExist anchor similarAnchors
 
-checkExternalResource :: VerifyConfig -> Text -> IO (VerifyResult VerifyError)
-checkExternalResource VerifyConfig{..} link
+checkExternalResource
+  :: VerifyConfig
+  -> IORef (Map Text $ Either VerifyError ())
+  -> Text
+  -> IO (VerifyResult VerifyError)
+checkExternalResource VerifyConfig{..} externalCheckedRef link
   | isIgnored = return mempty
   | doesReferLocalhost = return mempty
   | otherwise = fmap toVerifyRes $ do
-      makeRequest HEAD 0.3 >>= \case
-        Right () -> return $ Right ()
-        Left   _ -> makeRequest GET 0.7
+      externalChecked <- readIORef externalCheckedRef
+      if M.member link externalChecked
+      then return $ externalChecked M.! link
```

Comment:
`!` is a partial function, we should avoid it.

We should use `lookup` instead of `member`+`!`